enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -62,11 +61,14 @@
 
   Writes (even nested) yaml keys in a deterministic fashion."
   [filename obj]
-  (if (u.files/make-writable-parents! filename)
+  (io/make-parents filename)
+  (try
     (spit filename (yaml/generate-string (serialization-deep-sort obj)
                                          {:dumper-options {:flow-style :block :split-lines false}}))
-
-    (throw (ex-info (format "Destination path is not writeable: %s" filename) {:filename filename}))))
+    (catch Exception e
+      (if-not (.canWrite (.getParentFile (io/file filename)))
+        (throw (ex-info (format "Destination path is not writeable: %s" filename) {:filename filename}))
```

Comment:
This code path is untested and in practice seems really unlikely to be hit. The folder they're extracting to would have to be writable but the folders inside it like `collections` would have to be not writeable. Do you have a reason to keep this behaviour or can we remove it? Otherwise it should be tested

Reply:
On second thought, I guess it's possible and it's nice to have this otherwise it would be coupled to v2-dump!